.github/workflows/build.yml

```diff
@@ -4,98 +4,32 @@ on:
       - master
-  pull_request:
-    types:
-      - opened
-      - synchronize
-      - reopened
+  workflow_call:
+    inputs:
+      configuration:
+        type: string
+        default: Debug
 
 jobs:
   build:
     runs-on: ubuntu-latest
-
-    strategy:
-      fail-fast: false
-      matrix:
-        mono:
-          - latest
-          - '6.12'
-          - '6.10'
-          - '6.8'
-        configuration:
-          - Debug
-          - Release
-
-    container:
-      image: mono:${{ matrix.mono }}
-
     steps:
-      - name: Adding HTTPS support to APT for old Mono images
-        run: |
-          apt-get update || true
-          apt-get install -y apt-transport-https
-      - name: Installing checkout/build dependencies
-        run: apt-get update && apt-get install -y git
-      - uses: actions/checkout@v3
-
+      - uses: actions/checkout@v4
       - name: Setup .NET Core
-        uses: actions/setup-dotnet@v3
+        uses: actions/setup-dotnet@v4
         with:
           dotnet-version: '7'
-      - name: Install runtime dependencies
-        run: apt-get install -y xvfb
-      - name: Restore cache for _build/tools
-        uses: actions/cache@v3
-        with:
-          path: _build/tools
-          key: build-tools-${{ hashFiles('build', 'build.ps1', 'build.cake') }}
-      - name: Restore cache for _build/cake
-        uses: actions/cache@v3
-        with:
-          path: _build/cake
-          key: build-cake-${{ hashFiles('build.cake') }}
-      - name: Restore cache for _build/lib/nuget
-        uses: actions/cache@v3
-        with:
-          path: _build/lib/nuget
-          key: nuget-oldref-modules-${{ hashFiles('**/packages.config') }}-${{ hashFiles('**/*.csproj') }}
-      - name: Restore cache for ~/.nuget/packages
-        uses: actions/cache@v3
-        with:
-          path: ~/.nuget/packages
-          key: nuget-packref-modules-${{ hashFiles('**/packages.config') }}-${{ hashFiles('**/*.csproj') }}
-
       - name: Build ckan.exe and netkan.exe
-        run: ./build --configuration=${{ matrix.configuration}}
+        run: ./build --configuration=${{ inputs.configuration }}
       - name: Run tests
-        run: xvfb-run ./build test+only --configuration=${{ matrix.configuration }} --where="Category!=FlakyNetwork"
-      - name: Run inflator container smoke test
-        run: |
-          cd _build
-          curl -fsSL https://get.docker.com -o get-docker.sh
-          sh get-docker.sh
-          docker build --tag inflator --file ../Dockerfile.netkan .
-          docker run --rm --name inflator --entrypoint /bin/bash inflator -c "
-            mono netkan.exe https://raw.githubusercontent.com/KSP-CKAN/NetKAN/master/NetKAN/ZeroMiniAVC.netkan
-          "
-        if: matrix.configuration == 'release'
-
-      - name: Upload ckan.exe artifact
-        uses: actions/upload-artifact@v3
+        run: xvfb-run ./build test+only --configuration=${{ inputs.configuration }} --where="Category!=FlakyNetwork"
```

Comment:
I feel like `build.yml` should no longer run the tests here if we have an explicit `test.tml` file now.

Reply:
This tests it within our build step, and ensures that any time we need to consume it elsewhere we don't need to remember to test it.

What we could do is setting this to only run when the configuration is `Release`. As any `Debug` is expected to be run via the test workflow.

Reply:
Disabling it shaves off about 30 seconds, on balance I'm not sure if it's worth it. Whilst potentially unnecessary, it would shortcut the build by failing *before* doing the extended tests. Along with providing a strong guarantee that a test was run *before* being used for the release/deploy assets.

Reply:
There isn't really a point to having `test.yml` at all if this stays here. Any test failure will be caught here before `test.yml` runs; as far as I know, we've never had tests that only pass on some versions of Mono, so `test.yml` will not achieve anything for us.

The point of handling multiple versions of Mono was the **compilation/build** step, because it's possible that some build techniques may fail on older Mono versions. We're losing that now, which is fine with me because even the newest versions of Mono are old now, but when I see `build.yml` and `test.yml`, it seems pretty clear where the testing should happen.

Reply:
I don't fully agree with you here, but I'll ponder, because I get your point. I will counter though, part of it is to ensure that the built output (that we consume elsewhere) is guaranteed to have had tests run. IMO, those CPU cycles are worth the cost to have that guarantee, that we can depend that the artifacts are built successfully. Because I can assure, I have experienced the assumption that they passed, and lost way more time than the what an extra 30 seconds in a workflow has (4 hours time = 480 x 30 seconds).

Personally even if it's belts and bracers, I'd prefer to have the extra step. It's going to pretty obvious that the build failed either way, and we will know the build failed a step earlier than we would have otherwise.

Reply:
Let's just get rid of `test.yml`, then, if you want to **always** run the tests after building. It no longer serves a purpose.

Reply:
So no tests in the mono containers? If we're fine to drop them, that's ok with me. To me, those tests are for the artifacts we need elsewhere, rather extended testing, so it made sense to break them out separately and depend on a pre-validated build.

Reply:
See above; Mono version mattered for _**compiling/building**_ CKAN, not the tests.